src/metric/ackbag.rs

```diff
@@ -0,0 +1,104 @@
+use std::sync::{Arc, Mutex};
+use std::{thread, time};
+use util;
+use uuid::Uuid;
+
+#[derive(Clone)]
+pub struct SyncAckProps {
+    acks_received: usize
+}
+
+impl SyncAckProps {
+    pub fn new() -> Self {
+        SyncAckProps {
+            acks_received: 0,
+        }
+    }
+
+    pub fn ack(&mut self) {
+        self.acks_received.saturating_add(1);
+    }
+
+    pub fn acks_received(&self) -> usize {
+        self.acks_received
+    }
+}
+
+type SyncAckBagMap = util::HashMap<Uuid, SyncAckProps>;
+
+pub struct SyncAckBag {
+    waiting_syncs: Arc<Mutex<SyncAckBagMap>>
+}
+
+
+impl SyncAckBag {
+    /// Creates a SyncAckBag
+    pub fn new() -> Self {
```

Comment:
Recommend implementing Default for SyncAcBag versus having a blank new constructor.